teps/0048-task-results-without-results.md

```diff
@@ -426,6 +431,100 @@ with the git events such as `when a pull request is created`, etc but it
 doesn't runs. Having default results defined in `check-name-matches` Task
 will fix the `Pipeline`.
 
+## Proposal
+
+We propose adding an optional field - `default` - to the `Result` specification in `Task`.
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: task
+spec:
+  results:
+  - name: merge_status
+    description: whether to rebase or squash
+    type: string
+    default: rebase
+  - name: branches
+    description: branches in the repository
+    type: array
+    default:
+    - main
+    - v1alpha1
+    - v1beta1
+    - v1
+  - name: images
+    type: object
+    properties:
+      node:
+       type: string
+       default: "node:latest" 
+      gcloud:
+       type: string
+       default: "gcloud:latest"
```

Comment:
@ywluogg @chuangw6 @yaoxiaoqi - would appreciate your review on the default value for results